src/js/angular/resources/controllers.js

```diff
@@ -65,211 +66,215 @@ resourcesCtrl.controller('ResourcesCtrl', ['$scope', '$timeout', 'MonitoringRest
-            const activeRepository = $scope.getActiveRepository();
-            return MonitoringRestService.monitorQuery(activeRepository).then(function (response) {
-                return response.data;
-            }).catch(function (data) {
-                $scope.error = getError(data);
-                $scope.loader = false;
-            });
+        const getActiveQueryMonitor = function(activeRepository) {
+            return MonitoringRestService.monitorQuery(activeRepository)
+                .then(function(response) {
+                    return response.data;
+                });
         };
-        let structuresMonitorPoll;
-        const getStructuresMonitorData = function () {
+        const getStructuresMonitorData = function() {
             const activeRepository = $scope.getActiveRepository();
-            if ($scope.error) {
-                return;
-            } else if (!activeRepository) {
-                structuresMonitorPoll = $timeout(getStructuresMonitorData, POLLING_INTERVAL);
-                return;
+            if (!activeRepository) {
+                return Promise.resolve();
             }
-            MonitoringRestService.monitorStructures(activeRepository).then(function (response) {
-                const data = response.data;
-                if (data) {
-                    const timestamp = new Date();
 
-                    Object.values($scope.structuresMonitorData).forEach((chart) => {
-                        chart.addData(timestamp, data);
+            return MonitoringRestService.monitorStructures(activeRepository)
+                .then(function(response) {
+                    processResponse(response, (timestamp, data) => {
+                        Object.values($scope.structuresMonitorData).forEach((chart) => {
+                            chart.addData(timestamp, data);
+                        });
                     });
+                });
+        };
+        const getClusterMonitorData = function() {
+            return MonitoringRestService.monitorCluster()
+                .then(function(response) {
+                    processResponse(response, (timestamp, data) => $scope.clusterHealthChart.addData(timestamp, data));
+                });
+        };
 
-                    if ($scope.firstLoad) {
-                        $scope.firstLoad = false;
-
-                        const timer = $timeout(function () {
-                            $scope.loader = false;
-                        }, 500);
 
-                        $scope.$on('$destroy', function () {
-                            $timeout.cancel(timer);
-                        });
+        // TODO: Organize the following somehow
+        const getData = (monitor) => {
+            if (hasMonitorError(monitor.error)) {
+                if (monitor.error.retries === MAX_RETRIES) {
+                    if (monitor.poll) {
+                        $timeout.cancel(monitor.poll);
                     }
+                    return;
```

Comment:
I'm a bit confused by this. In the other branch, this function is supposed to return a Promise, which means that you has to handle this return somehow.